README.md

```diff
@@ -2,3 +2,79 @@
 
 This repository contains examples in various languages that demonstrate how to
 build and test software with the EngFlow Remote Execution service.
+
+## Local execution
+### Building locally
+
+- Build each example with  
+
+  `bazel build //${DIRECTORY}/...`  
+
+  for `cpp`,  `csharp`, `docker`, `genrules`, `go`, `java`, `kotlin`, `perl`, `python`, `scala` and `typescript` or build them all with  
+  
+  `bazel build //...`
+- Build `swift` example with  
+  
+  `bazel build //swift:test --config=clang` 
+  
+  make sure `clang` is in your `PATH`.
+
+### Testing locally
+
+- Test each example with
+
+  `bazel test //${DIRECTORY}/...`  
+
+  for `java`, `scala` and `typescript`.
+
+- Test `swift` example with
+
+  `bazel test //swift:test --config=clang`
+
+- Test `python` example with
+
+  `bazel run //python:requirements.update && bazel test //python/...`
+
+## Remote execution
+
+Make sure to include `--remote_executor` and `--bes_backend`, in your `remote` configuration, as well as access credentials if needed. For instance,
+
+```bzl
+build:remote --remote_executor=grpcs://${CLUSTER_ENDPOINT}/
+build:remote --bes_backend=grpcs://${CLUSTER_ENDPOINT}/
+build:remote --bes_results_url=https://${CLUSTER_ENDPOINT}/invocation/
+build:remote --nogoogle_default_credentials
+build:remote --bes_lifecycle_events
+build:remote --tls_client_certificate=/path/to/credentials/cert.crt
+build:remote --tls_client_key=/path/to/credentials/cert.key
+```
+### Building on remote
+
+
+- Build each example with  
+
+  `bazel build //${DIRECTORY}/... --config=remote --enable_platform_specific_config` 
```

Comment:
Is `enable_platform_specific_config` actually needed? (It's in `.bazelrc` already, but more importantly, we don't seem to use it anywhere... unless I'm missing something.)

Reply:
In fact, as it is used `build --enable_platform_specific_config` in the `.bazelrc` so it is used by default when building locally. However, when using `--config=remote` it only uses the `remote` configuration. The `remote` configuration for a cluster, for instance `opal`, is placed in the `.bazelrc.user` file and it may or not include the use `--enable_platform_specific_config` which needs to be activated for remote too.

Reply:
Actually, `--config` flags are additive, so using `--config=remote` adds the `remote` configuration's flags on top of the defaults. So `--enable_platform_specific_config` gets added anyway.

(And I've seen that we do have platform-specific config - `build:macos` and `build:linux` in `.bazelrc`; missed that before)

Reply:
I was thinking the same when I was adding the flag, but if I do not add it to `bazel build //... --config=opal`, it will simply fail. Using `--enable_platform_specific_config` makes the example use the correct toolchains.

Reply:
I'm confused; see for example on [this invocation](https://oss-cluster.engflow.com/invocation/67334039-f08e-4d7d-a59e-ca15abc41b47#details) - my command line is represented accurately (it didn't have this flag explicitly), yet it appears (correctly) in the list of flags. It's even the first one.

This is not a big deal, but I think it's better to have shorter, more concise command lines in examples. Let's follow up offline if this isn't working for you to see why.